File: 5. Array Delete And Shift - GFG/5.-array-delete-and-shift.cpp
// { Driver Code Starts
//Initial Template for C++


#include<bits/stdc++.h>
using namespace std;


 // } Driver Code Ends
//User function Template for C++


//Complete the function
void deleteFromArray(int arr[], int n, int index){
    
  //Your code here
  // directly shift all elements left by 1 place till index
  for(int i = index; i < n; i++){
      arr[i] = arr[i+1];
  }
  // setting last element to zero
  arr[n-1] = 0;
}


// { Driver Code Starts.


int main() {
	int t;
	cin>>t;
	while(t--)
	{
    	int n;
    	int index;
    	cin >> n;
    	cin >> index;
    	
    	int arr[n];
    	
    	for(int i = 0;i<n;i++){
    	    arr[i] = i+1;
    	}
    	
    	deleteFromArray(arr, n, index);
    
    	for(int i = 0;i<n;i++)
    	    cout << arr[i] << " ";
    	    
    	  cout<<endl;
	}
	return 0;
}  // } Driver Code Ends

File: Array/5. Array Delete And Shift - GFG/5.-array-delete-and-shift.py
#User function Template for python3

def deleteFromArray(arr,n,idx):
    #code here
    # shifting all element sleft by 1 till idx
    for i in range(idx,n-1):
        arr[i] = arr[i+1]
    arr[n-1] = 0 #insert  0 at the end 

#{ 
#  Driver Code Starts
#Initial Template for Python 3

#CONTRIBUTED BY RavinderSinghPB
if __name__ == '__main__':
    tcs= int(input())
    
    for _ in range(tcs):
        n=int(input())
        idx=int(input())
        
        arr=[i+1 for i in range(n)]
        
        deleteFromArray(arr,n,idx)
        
        for e in arr:
            print(e,end=' ')
        print()
        
# } Driver Code Ends